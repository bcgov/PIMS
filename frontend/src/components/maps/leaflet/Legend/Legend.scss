@import '../../../../colors.scss';

.legend-control {
  &.card {
    min-width: 200px;
    .card-header {
      margin-bottom: 0;
      border-bottom: none;
      background-color: #fff;
      font-weight: 900;
      color: $text-color;
    }
    .card-body {
      padding-top: 0px;
      padding-bottom: 10px;
      .row {
        margin-top: 5px;
        padding-bottom: 5px;
        .col {
          display: flex;
          align-items: center;
          color: $text-color;
        }
      }
    }
  }
}
