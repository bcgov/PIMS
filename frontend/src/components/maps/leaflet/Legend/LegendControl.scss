@import '../../../../styles.scss';

.legendTooltip {
  .arrow::before {
    border-right-color: white !important;
    box-shadow: 2px grey;
  }

  .tooltip-inner {
    background-color: white;
    color: black;
    box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
    padding: 0px;
    text-align: left;
    .row {
      .label {
        padding-left: 0;
        display: flex;
        align-items: center;
      }
    }
  }
}
