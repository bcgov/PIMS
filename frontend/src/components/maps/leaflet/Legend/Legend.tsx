import * as React from 'react';
import { Row, Col, Card, Image } from 'react-bootstrap';
import './Legend.scss';

export const Legend = () => {
  const keys = React.useMemo(() => {
    return [
      {
        pin: require('assets/images/pins/land-reg.png'),
        label: 'Parcel',
      },
      {
        pin: require('assets/images/pins/building-reg.png'),
        label: 'Building',
      },
      {
        pin: require('assets/images/pins/subdiv-reg.png'),
        label: 'Proposed Subdivision',
      },
      {
        pin: require('assets/images/pins/marker-red.png'),
        label: 'Enhanced Referral Process',
      },
      {
        pin: require('assets/images/pins/marker-purple.png'),
        label: 'Surplus Properties List',
      },
    ];
  }, []);

  return (
    <Card className="legend-control">
      <Card.Header>Marker Legend</Card.Header>

      <Card.Body>
        {keys.map((item, index) => {
          return (
            <Row key={index}>
              <Col xs={2}>
                <Image height={25} src={item.pin} />
              </Col>
              <Col className="label">{item.label}</Col>
            </Row>
          );
        })}
      </Card.Body>
    </Card>
  );
};
