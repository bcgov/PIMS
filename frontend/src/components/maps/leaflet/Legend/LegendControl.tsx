import './LegendControl.scss';

import variables from '_variables.module.scss';
import TooltipWrapper from 'components/common/TooltipWrapper';
import * as React from 'react';
import { Button, Overlay, Tooltip } from 'react-bootstrap';
import ClickAwayListener from 'react-click-away-listener';
import { FiMapPin } from 'react-icons/fi';
import Control from 'react-leaflet-control';
import styled from 'styled-components';

import { Legend } from './Legend';

const LegendButton = styled(Button)`
  background-color: #ffffff !important;
  color: ${variables.darkVariantColor} !important;
  width: 40px;
  height: 40px;
  font-size: 25px;
  display: flex;
  align-items: center;
`;

export const LegendControl: React.FC = () => {
  const [visible, setVisible] = React.useState<boolean>(false);
  const target = React.useRef(null);

  return (
    <Control position="topleft">
      <ClickAwayListener onClickAway={() => setVisible(false)}>
        <TooltipWrapper toolTipId="marker-legendId" toolTip={visible ? undefined : 'Marker legend'}>
          <LegendButton ref={target} onClick={() => setVisible(!visible)}>
            <FiMapPin />
          </LegendButton>
        </TooltipWrapper>
        <Overlay target={target.current!} show={visible} placement="right">
          {(props: any) => {
            return (
              <Tooltip id="overlay-legend" {...props} show={`${visible}`} className="legendTooltip">
                <Legend />
              </Tooltip>
            );
          }}
        </Overlay>
      </ClickAwayListener>
    </Control>
  );
};
