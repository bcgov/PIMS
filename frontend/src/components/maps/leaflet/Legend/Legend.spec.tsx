import React from 'react';
import renderer from 'react-test-renderer';

import { Legend } from './Legend';

describe('Map Legend View', () => {
  it('Legend renders correctly', () => {
    const tree = renderer.create(<Legend />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
