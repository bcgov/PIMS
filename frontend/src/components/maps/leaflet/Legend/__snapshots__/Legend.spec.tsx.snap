// Vitest Snapshot v1

exports[`Map Legend View > Legend renders correctly 1`] = `
<div
  className="legend-control card"
>
  <div
    className="card-header"
  >
    Marker Legend
  </div>
  <div
    className="card-body"
  >
    <div
      className="row"
    >
      <div
        className="col-2"
      >
        <img
          className=""
          height={25}
          src="/src/assets/images/pins/land-reg.png"
        />
      </div>
      <div
        className="label col"
      >
        Parcel
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-2"
      >
        <img
          className=""
          height={25}
          src="/src/assets/images/pins/building-reg.png"
        />
      </div>
      <div
        className="label col"
      >
        Building
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-2"
      >
        <img
          className=""
          height={25}
          src="/src/assets/images/pins/subdiv-reg.png"
        />
      </div>
      <div
        className="label col"
      >
        Proposed Subdivision
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-2"
      >
        <img
          className=""
          height={25}
          src="/src/assets/images/pins/marker-red.png"
        />
      </div>
      <div
        className="label col"
      >
        Enhanced Referral Process
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-2"
      >
        <img
          className=""
          height={25}
          src="/src/assets/images/pins/marker-purple.png"
        />
      </div>
      <div
        className="label col"
      >
        Surplus Properties List
      </div>
    </div>
  </div>
</div>
`;
